Errors in mypy test locally

### Describe the bug

When run locally 
`make mypy`

I am getting many errors from augmentation package

```

===================================================================== FAILURES ======================================================================
___________________________________________________________________ test session ____________________________________________________________________
mypy exited with status 1.
________________________________________________________ kornia/augmentation/_2d/mix/base.py ________________________________________________________
252: error: Dict entry 0 has incompatible
252: error: Dict entry 1 has incompatible
_______________________________________________________ kornia/augmentation/_2d/mix/cutmix.py _______________________________________________________
225: error: No overload variant of "to"
225: note: Possible overload variants:
225: note:     def to(self, dtype: dtype, non_blocking: bool = ..., copy: bool = ...) -> Tensor
225: note:     def to(self, device: Union[device, str, None] = ..., dtype: Optional[dtype] = ..., non_blocking: bool = ..., copy: bool = ...) -> Tensor
225: note:     def to(self, other: Tensor, non_blocking: bool = ..., copy: bool = ...) -> Tensor
226: error: No overload variant of "to"
226: note: Possible overload variants:
226: note:     def to(self, dtype: dtype, non_blocking: bool = ..., copy: bool = ...) -> Tensor
226: note:     def to(self, device: Union[device, str, None] = ..., dtype: Optional[dtype] = ..., non_blocking: bool = ..., copy: bool = ...) -> Tensor
226: note:     def to(self, other: Tensor, non_blocking: bool = ..., copy: bool = ...) -> Tensor
239: error: No overload variant of
239: note: Possible overload variants:
239: note:     def zeros(size: Union[Size, List[int], Tuple[int, ...]], *, names: Optional[Sequence[Union[str, ellipsis, None]]], dtype: Optional[dtype] = ..., layout: Optional[layout] = ..., device: Union[device, str, None] = ..., pin_memory: Optional[bool] = ..., requires_grad: Optional[bool] = ...) -> Tensor
239: note:     def zeros(*size: int, names: Optional[Sequence[Union[str, ellipsis, None]]], dtype: Optional[dtype] = ..., layout: Optional[layout] = ..., device: Union[device, str, None] = ..., pin_memory: Optional[bool] = ..., requires_grad: Optional[bool] = ...) -> Tensor
239: note:     def zeros(size: Sequence[Union[int, SymInt]], *, out: Optional[Tensor] = ..., dtype: Optional[dtype] = ..., layout: Optional[layout] = ..., device: Union[device, str, None] = ..., pin_memory: Optional[bool] = ..., requires_grad: Optional[bool] = ...) -> Tensor
239: note:     def zeros(*size: int, out: Optional[Tensor] = ..., dtype: Optional[dtype] = ..., layout: Optional[layout] = ..., device: Union[device, str, None] = ..., pin_memory: Optional[bool] = ..., requires_grad: Optional[bool] = ...) -> Tensor
240: error: Item "None" of
240: error: Argument 1 to "range" has
244: error: No overload variant of "to"
244: note: Possible overload variants:
244: note:     def to(self, dtype: dtype, non_blocking: bool = ..., copy: bool = ...) -> Tensor
244: note:     def to(self, device: Union[device, str, None] = ..., dtype: Optional[dtype] = ..., non_blocking: bool = ..., copy: bool = ...) -> Tensor
244: note:     def to(self, other: Tensor, non_blocking: bool = ..., copy: bool = ...) -> Tensor
245: error: No overload variant of "to"
245: note: Possible overload variants:
245: note:     def to(self, dtype: dtype, non_blocking: bool = ..., copy: bool = ...) -> Tensor
245: note:     def to(self, device: Union[device, str, None] = ..., dtype: Optional[dtype] = ..., non_blocking: bool = ..., copy: bool = ...) -> Tensor
245: note:     def to(self, other: Tensor, non_blocking: bool = ..., copy: bool = ...) -> Tensor
_______________________________________________________ kornia/augmentation/_2d/mix/mixup.py ________________________________________________________
205: error: No overload variant of "to" of
205: note: Possible overload variants:
205: note:     def to(self, dtype: dtype, non_blocking: bool = ..., copy: bool = ...) -> Tensor
205: note:     def to(self, device: Union[device, str, None] = ..., dtype: Optional[dtype] = ..., non_blocking: bool = ..., copy: bool = ...) -> Tensor
205: note:     def to(self, other: Tensor, non_blocking: bool = ..., copy: bool = ...) -> Tensor
206: error: No overload variant of "to" of
206: note: Possible overload variants:
206: note:     def to(self, dtype: dtype, non_blocking: bool = ..., copy: bool = ...) -> Tensor
206: note:     def to(self, device: Union[device, str, None] = ..., dtype: Optional[dtype] = ..., non_blocking: bool = ..., copy: bool = ...) -> Tensor
206: note:     def to(self, other: Tensor, non_blocking: bool = ..., copy: bool = ...) -> Tensor
207: error: No overload variant of "zeros"
207: note: Possible overload variants:
207: note:     def zeros(size: Union[Size, List[int], Tuple[int, ...]], *, names: Optional[Sequence[Union[str, ellipsis, None]]], dtype: Optional[dtype] = ..., layout: Optional[layout] = ..., device: Union[device, str, None] = ..., pin_memory: Optional[bool] = ..., requires_grad: Optional[bool] = ...) -> Tensor
207: note:     def zeros(*size: int, names: Optional[Sequence[Union[str, ellipsis, None]]], dtype: Optional[dtype] = ..., layout: Optional[layout] = ..., device: Union[device, str, None] = ..., pin_memory: Optional[bool] = ..., requires_grad: Optional[bool] = ...) -> Tensor
207: note:     def zeros(size: Sequence[Union[int, SymInt]], *, out: Optional[Tensor] = ..., dtype: Optional[dtype] = ..., layout: Optional[layout] = ..., device: Union[device, str, None] = ..., pin_memory: Optional[bool] = ..., requires_grad: Optional[bool] = ...) -> Tensor
207: note:     def zeros(*size: int, out: Optional[Tensor] = ..., dtype: Optional[dtype] = ..., layout: Optional[layout] = ..., device: Union[device, str, None] = ..., pin_memory: Optional[bool] = ..., requires_grad: Optional[bool] = ...) -> Tensor
220: error: No overload variant of "to" of
220: note: Possible overload variants:
220: note:     def to(self, dtype: dtype, non_blocking: bool = ..., copy: bool = ...) -> Tensor
220: note:     def to(self, device: Union[device, str, None] = ..., dtype: Optional[dtype] = ..., non_blocking: bool = ..., copy: bool = ...) -> Tensor
220: note:     def to(self, other: Tensor, non_blocking: bool = ..., copy: bool = ...) -> Tensor
221: error: No overload variant of "to" of
221: note: Possible overload variants:
221: note:     def to(self, dtype: dtype, non_blocking: bool = ..., copy: bool = ...) -> Tensor
221: note:     def to(self, device: Union[device, str, None] = ..., dtype: Optional[dtype] = ..., non_blocking: bool = ..., copy: bool = ...) -> Tensor
221: note:     def to(self, other: Tensor, non_blocking: bool = ..., copy: bool = ...) -> Tensor
222: error: No overload variant of "to" of
222: note: Possible overload variants:
222: note:     def to(self, dtype: dtype, non_blocking: bool = ..., copy: bool = ...) -> Tensor
222: note:     def to(self, device: Union[device, str, None] = ..., dtype: Optional[dtype] = ..., non_blocking: bool = ..., copy: bool = ...) -> Tensor
222: note:     def to(self, other: Tensor, non_blocking: bool = ..., copy: bool = ...) -> Tensor
___________________________________________________ kornia/augmentation/random_generator/base.py ____________________________________________________
38: error: Incompatible types in
39: error: Incompatible types in
________________________________________________ kornia/augmentation/random_generator/_2d/resize.py _________________________________________________
35: error: Incompatible
36: error: Incompatible
======================================================================= mypy ========================================================================
assignment (expression has type "device", variable has type "None") 
[assignment]
                self.device = device
                              ^
assignment (expression has type "dtype", variable has type "None")  [assignment]
                self.dtype = dtype
                             ^
types in assignment (expression has type "Optional[device]", base class
"RandomGeneratorBase" defined the type as "None")  [assignment]
            self.device: Optional[torch.device] = None
                                                  ^
types in assignment (expression has type "Optional[dtype]", base class
"RandomGeneratorBase" defined the type as "None")  [assignment]
            self.dtype: Optional[torch.dtype] = None
                                                ^
type "str": "device"; expected "str": "Tensor"  [dict-item]
                self._params.update({"device": in_tensor.device, "dtype": ...
                                     ^
type "str": "dtype"; expected "str": "Tensor"  [dict-item]
    ...self._params.update({"device": in_tensor.device, "dtype": in_tensor.dt...
                                                        ^
"_TensorBase" matches argument types "device", "Tensor"  [call-overload]
                    input.to(device=input.device, dtype=params["dtype"]),
                    ^
"_TensorBase" matches argument types "device", "Tensor"  [call-overload]
                    input.to(device=input.device, dtype=params["dtype"]),
                    ^
matches argument types "Tuple[int]", "device", "Tensor"  [call-overload]
                    torch.zeros((len(input),), device=input.device, dtype=...
                    ^
"_TensorBase" matches argument types "device", "Tensor"  [call-overload]
                    input.to(device=input.device, dtype=params["dtype"]),
                    ^
"_TensorBase" matches argument types "device", "Tensor"  [call-overload]
                    labels_permute.to(device=input.device, dtype=params["d...
                    ^
"_TensorBase" matches argument types "device", "Tensor"  [call-overload]
                    params["mixup_lambdas"].to(device=input.device, dtype=...
                    ^
of "_TensorBase" matches argument types "device", "Tensor"  [call-overload]
                            input.to(device=input.device, dtype=params["dt...
                            ^
of "_TensorBase" matches argument types "device", "Tensor"  [call-overload]
                            labels_permute.to(device=input.device, dtype=p...
                            ^
"zeros" matches argument types "int", "device", "Tensor"  [call-overload]
            lam = torch.zeros((len(input)), device=input.device, dtype=par...
                  ^
"Optional[RandomGeneratorBase]" has no attribute "num_mix"  [union-attr]
            for _ in range(self._param_generator.num_mix):
                           ^
incompatible type "Union[Tensor, Module, Any]"; expected "SupportsIndex" 
[arg-type]
            for _ in range(self._param_generator.num_mix):
                           ^
of "_TensorBase" matches argument types "device", "Tensor"  [call-overload]
                            input.to(device=input.device, dtype=params["dt...
                            ^
of "_TensorBase" matches argument types "device", "Tensor"  [call-overload]
                            input.to(device=input.device, dtype=params["dt...
                            ^
Found 19 errors in 5 files (checked 272 source files)
============================================================== short test summary info ==============================================================
FAILED kornia/__init__.py::mypy-status
FAILED kornia/augmentation/_2d/mix/base.py::mypy
FAILED kornia/augmentation/_2d/mix/cutmix.py::mypy
FAILED kornia/augmentation/_2d/mix/mixup.py::mypy
FAILED kornia/augmentation/random_generator/base.py::mypy
FAILED kornia/augmentation/random_generator/_2d/resize.py::mypy

```


### Reproduction steps

```bash
make mypy
```


### Expected behavior

No error

### Environment

```shell
Versions

Collecting environment information...
PyTorch version: 1.13.0.dev20220916
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 12.6 (arm64)
GCC version: Could not collect
Clang version: 14.0.6
CMake version: version 3.22.3
Libc version: N/A

Python version: 3.9.4 | packaged by conda-forge | (default, May 10 2021, 22:10:52) [Clang 11.1.0 ] (64-bit runtime)
Python platform: macOS-12.6-arm64-arm-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] mypy==0.910
[pip3] mypy-extensions==0.4.3
[pip3] numpy==1.21.2
[pip3] pytest-mypy==0.9.0
[pip3] torch==1.13.0.dev20220916
[pip3] torch-dimcheck==0.0.1
[pip3] torchaudio==0.11.0
[pip3] torchvision==0.10.0
[conda] numpy 1.21.2 pypi_0 pypi
[conda] torch 1.13.0.dev20220916 pypi_0 pypi
[conda] torch-dimcheck 0.0.1 pypi_0 pypi
[conda] torchaudio 0.11.0 pypi_0 pypi
[conda] torchvision 0.10.0 pypi_0 pypi
```


### Additional context

_No response_

--- a/kornia/augmentation/_2d/mix/base.py
+++ b/kornia/augmentation/_2d/mix/base.py
@@ -5,7 +5,7 @@ import torch
 
 from kornia.augmentation.base import TensorWithTransformMat, _BasicAugmentationBase
 from kornia.augmentation.utils import _transform_input, _transform_output_shape, _validate_input_dtype
-from kornia.constants import DataKey
+from kornia.constants import DataKey, DType
 from kornia.core import Tensor
 from kornia.geometry.boxes import Boxes
 from kornia.testing import KORNIA_UNWRAP
@@ -249,7 +249,7 @@ class MixAugmentationBaseV2(_BasicAugmentationBase):
             in_tensor: Tensor = input[in_tensor_idx]
             in_tensor = self.transform_tensor(in_tensor)
             self._params = self.forward_parameters(in_tensor.shape)
-            self._params.update({"device": in_tensor.device, "dtype": in_tensor.dtype})
+            self._params.update({"dtype": torch.tensor(DType.get(in_tensor.dtype).value)})
         else:
             self._params = params
 

--- a/kornia/augmentation/_2d/mix/cutmix.py
+++ b/kornia/augmentation/_2d/mix/cutmix.py
@@ -222,9 +222,9 @@ class RandomCutMixV2(MixAugmentationBaseV2):
             out_labels.append(
                 torch.stack(
                     [
-                        input.to(device=input.device, dtype=DType.to_torch(params["dtype"].item())),
-                        labels_permute.to(device=input.device, dtype=DType.to_torch(params["dtype"].item())),
-                        lam.to(device=input.device, dtype=DType.to_torch(params["dtype"].item())),
+                        input.to(device=input.device, dtype=DType.to_torch(int(params["dtype"].item()))),
+                        labels_permute.to(device=input.device, dtype=DType.to_torch(int(params["dtype"].item()))),
+                        lam.to(device=input.device, dtype=DType.to_torch(int(params["dtype"].item()))),
                     ],
                     dim=1,
                 )
@@ -236,13 +236,13 @@ class RandomCutMixV2(MixAugmentationBaseV2):
         self, input: Tensor, params: Dict[str, Tensor], flags: Optional[Dict[str, Any]] = None
     ) -> Tensor:
         out_labels = []
-        lam = torch.zeros((len(input)), device=input.device, dtype=DType.to_torch(params["dtype"].item()))
+        lam = torch.zeros((len(input)), device=input.device, dtype=DType.to_torch(int(params["dtype"].item())))
         for _ in range(self._param_generator.num_mix):
             out_labels.append(
                 torch.stack(
                     [
-                        input.to(device=input.device, dtype=DType.to_torch(params["dtype"].item())),
-                        input.to(device=input.device, dtype=DType.to_torch(params["dtype"].item())),
+                        input.to(device=input.device, dtype=DType.to_torch(int(params["dtype"].item()))),
+                        input.to(device=input.device, dtype=DType.to_torch(int(params["dtype"].item()))),
                         lam,
                     ],
                     dim=1,

--- a/kornia/augmentation/_2d/mix/cutmix.py
+++ b/kornia/augmentation/_2d/mix/cutmix.py
@@ -222,9 +222,9 @@ class RandomCutMixV2(MixAugmentationBaseV2):
             out_labels.append(
                 torch.stack(
                     [
-                        input.to(device=input.device, dtype=DType.to_torch(params["dtype"])),
-                        labels_permute.to(device=input.device, dtype=DType.to_torch(params["dtype"])),
-                        lam.to(device=input.device, dtype=DType.to_torch(params["dtype"])),
+                        input.to(device=input.device, dtype=DType.to_torch(params["dtype"].item())),
+                        labels_permute.to(device=input.device, dtype=DType.to_torch(params["dtype"].item())),
+                        lam.to(device=input.device, dtype=DType.to_torch(params["dtype"].item())),
                     ],
                     dim=1,
                 )
@@ -236,13 +236,13 @@ class RandomCutMixV2(MixAugmentationBaseV2):
         self, input: Tensor, params: Dict[str, Tensor], flags: Optional[Dict[str, Any]] = None
     ) -> Tensor:
         out_labels = []
-        lam = torch.zeros((len(input)), device=input.device, dtype=DType.to_torch(params["dtype"]))
+        lam = torch.zeros((len(input)), device=input.device, dtype=DType.to_torch(params["dtype"].item()))
         for _ in range(self._param_generator.num_mix):
             out_labels.append(
                 torch.stack(
                     [
-                        input.to(device=input.device, dtype=DType.to_torch(params["dtype"])),
-                        input.to(device=input.device, dtype=DType.to_torch(params["dtype"])),
+                        input.to(device=input.device, dtype=DType.to_torch(params["dtype"].item())),
+                        input.to(device=input.device, dtype=DType.to_torch(params["dtype"].item())),
                         lam,
                     ],
                     dim=1,

--- a/kornia/augmentation/_2d/mix/cutmix.py
+++ b/kornia/augmentation/_2d/mix/cutmix.py
@@ -6,7 +6,7 @@ import torch
 from kornia.augmentation import random_generator as rg
 from kornia.augmentation._2d.mix.base import MixAugmentationBase, MixAugmentationBaseV2
 from kornia.augmentation.utils import _shape_validation
-from kornia.constants import DataKey
+from kornia.constants import DataKey, DType
 from kornia.core import Tensor
 from kornia.geometry.bbox import bbox_to_mask, infer_bbox_shape
 
@@ -222,9 +222,9 @@ class RandomCutMixV2(MixAugmentationBaseV2):
             out_labels.append(
                 torch.stack(
                     [
-                        input.to(device=input.device, dtype=params["dtype"]),
-                        labels_permute.to(device=input.device, dtype=params["dtype"]),
-                        lam.to(device=input.device, dtype=params["dtype"]),
+                        input.to(device=input.device, dtype=DType.to_torch(params["dtype"])),
+                        labels_permute.to(device=input.device, dtype=DType.to_torch(params["dtype"])),
+                        lam.to(device=input.device, dtype=DType.to_torch(params["dtype"])),
                     ],
                     dim=1,
                 )
@@ -236,13 +236,13 @@ class RandomCutMixV2(MixAugmentationBaseV2):
         self, input: Tensor, params: Dict[str, Tensor], flags: Optional[Dict[str, Any]] = None
     ) -> Tensor:
         out_labels = []
-        lam = torch.zeros((len(input)), device=input.device, dtype=params["dtype"])
+        lam = torch.zeros((len(input)), device=input.device, dtype=DType.to_torch(params["dtype"]))
         for _ in range(self._param_generator.num_mix):
             out_labels.append(
                 torch.stack(
                     [
-                        input.to(device=input.device, dtype=params["dtype"]),
-                        input.to(device=input.device, dtype=params["dtype"]),
+                        input.to(device=input.device, dtype=DType.to_torch(params["dtype"])),
+                        input.to(device=input.device, dtype=DType.to_torch(params["dtype"])),
                         lam,
                     ],
                     dim=1,

--- a/kornia/augmentation/_2d/mix/mixup.py
+++ b/kornia/augmentation/_2d/mix/mixup.py
@@ -5,7 +5,7 @@ import torch
 
 from kornia.augmentation import random_generator as rg
 from kornia.augmentation._2d.mix.base import MixAugmentationBase, MixAugmentationBaseV2
-from kornia.constants import DataKey
+from kornia.constants import DataKey, DType
 from kornia.core import Tensor
 
 
@@ -202,9 +202,9 @@ class RandomMixUpV2(MixAugmentationBaseV2):
     ) -> Tensor:
         out_labels = torch.stack(
             [
-                input.to(device=input.device, dtype=params["dtype"]),
-                input.to(device=input.device, dtype=params["dtype"]),
-                torch.zeros((len(input),), device=input.device, dtype=params["dtype"]),
+                input.to(device=input.device, dtype=DType.to_torch(params["dtype"])),
+                input.to(device=input.device, dtype=DType.to_torch(params["dtype"])),
+                torch.zeros((len(input),), device=input.device, dtype=DType.to_torch(params["dtype"])),
             ],
             dim=-1,
         )
@@ -217,9 +217,9 @@ class RandomMixUpV2(MixAugmentationBaseV2):
 
         out_labels = torch.stack(
             [
-                input.to(device=input.device, dtype=params["dtype"]),
-                labels_permute.to(device=input.device, dtype=params["dtype"]),
-                params["mixup_lambdas"].to(device=input.device, dtype=params["dtype"]),
+                input.to(device=input.device, dtype=DType.to_torch(params["dtype"])),
+                labels_permute.to(device=input.device, dtype=DType.to_torch(params["dtype"])),
+                params["mixup_lambdas"].to(device=input.device, dtype=DType.to_torch(params["dtype"])),
             ],
             dim=-1,
         )

--- a/kornia/augmentation/_2d/mix/mixup.py
+++ b/kornia/augmentation/_2d/mix/mixup.py
@@ -202,9 +202,9 @@ class RandomMixUpV2(MixAugmentationBaseV2):
     ) -> Tensor:
         out_labels = torch.stack(
             [
-                input.to(device=input.device, dtype=DType.to_torch(params["dtype"].item())),
-                input.to(device=input.device, dtype=DType.to_torch(params["dtype"].item())),
-                torch.zeros((len(input),), device=input.device, dtype=DType.to_torch(params["dtype"].item())),
+                input.to(device=input.device, dtype=DType.to_torch(int(params["dtype"].item()))),
+                input.to(device=input.device, dtype=DType.to_torch(int(params["dtype"].item()))),
+                torch.zeros((len(input),), device=input.device, dtype=DType.to_torch(int(params["dtype"].item()))),
             ],
             dim=-1,
         )
@@ -217,9 +217,9 @@ class RandomMixUpV2(MixAugmentationBaseV2):
 
         out_labels = torch.stack(
             [
-                input.to(device=input.device, dtype=DType.to_torch(params["dtype"].item())),
-                labels_permute.to(device=input.device, dtype=DType.to_torch(params["dtype"].item())),
-                params["mixup_lambdas"].to(device=input.device, dtype=DType.to_torch(params["dtype"].item())),
+                input.to(device=input.device, dtype=DType.to_torch(int(params["dtype"].item()))),
+                labels_permute.to(device=input.device, dtype=DType.to_torch(int(params["dtype"].item()))),
+                params["mixup_lambdas"].to(device=input.device, dtype=DType.to_torch(int(params["dtype"].item()))),
             ],
             dim=-1,
         )

--- a/kornia/augmentation/_2d/mix/mixup.py
+++ b/kornia/augmentation/_2d/mix/mixup.py
@@ -202,9 +202,9 @@ class RandomMixUpV2(MixAugmentationBaseV2):
     ) -> Tensor:
         out_labels = torch.stack(
             [
-                input.to(device=input.device, dtype=DType.to_torch(params["dtype"])),
-                input.to(device=input.device, dtype=DType.to_torch(params["dtype"])),
-                torch.zeros((len(input),), device=input.device, dtype=DType.to_torch(params["dtype"])),
+                input.to(device=input.device, dtype=DType.to_torch(params["dtype"].item())),
+                input.to(device=input.device, dtype=DType.to_torch(params["dtype"].item())),
+                torch.zeros((len(input),), device=input.device, dtype=DType.to_torch(params["dtype"].item())),
             ],
             dim=-1,
         )
@@ -217,9 +217,9 @@ class RandomMixUpV2(MixAugmentationBaseV2):
 
         out_labels = torch.stack(
             [
-                input.to(device=input.device, dtype=DType.to_torch(params["dtype"])),
-                labels_permute.to(device=input.device, dtype=DType.to_torch(params["dtype"])),
-                params["mixup_lambdas"].to(device=input.device, dtype=DType.to_torch(params["dtype"])),
+                input.to(device=input.device, dtype=DType.to_torch(params["dtype"].item())),
+                labels_permute.to(device=input.device, dtype=DType.to_torch(params["dtype"].item())),
+                params["mixup_lambdas"].to(device=input.device, dtype=DType.to_torch(params["dtype"].item())),
             ],
             dim=-1,
         )

--- a/kornia/constants.py
+++ b/kornia/constants.py
@@ -2,8 +2,9 @@ from enum import Enum, EnumMeta
 from typing import TypeVar, Union, cast
 
 import torch
+from torch import Tensor
 
-__all__ = ['pi', 'Resample', 'BorderType', 'SamplePadding']
+__all__ = ['pi', 'DType', 'Resample', 'BorderType', 'SamplePadding']
 
 pi = torch.tensor(3.14159265358979323846)
 T = TypeVar('T', bound='ConstantBase')
@@ -51,6 +52,42 @@ class SamplePadding(ConstantBase, Enum, metaclass=EnumMetaFlags):
     ZEROS = 0
     BORDER = 1
     REFLECTION = 2
+
+
+class DType(ConstantBase, Enum, metaclass=EnumMetaFlags):
+    INT64 = 0
+    FLOAT16 = 1
+    FLOAT32 = 2
+    FLOAT64 = 3
+
+    @classmethod
+    def get(cls, value: Union[str, int, torch.dtype, Tensor, T]) -> T:  # type: ignore
+        if type(value) is torch.dtype:
+            value = str(value).upper()  # Convert to str
+        if type(value) is Tensor:
+            value = value.item()  # Convert to int
+        if type(value) is str:
+            if value.upper().startswith("TORCH."):
+                return cls[value.upper()[6:]]  # type: ignore
+            return cls[value.upper()]  # type: ignore
+        if type(value) is int:
+            return cls(value)  # type: ignore
+        if type(value) is cls:
+            return value  # type: ignore
+        raise TypeError(f"Invalid identifier {value}.")
+
+    @classmethod
+    def to_torch(cls, value: Union[str, int, T]) -> T:  # type: ignore
+        data = cls.get(value=value)
+        if data == DType.INT64:
+            return torch.long
+        if data == DType.FLOAT16:
+            return torch.float16
+        if data == DType.FLOAT32:
+            return torch.float32
+        if data == DType.FLOAT64:
+            return torch.float64
+        raise ValueError()
 
 
 # TODO: (low-priority) add INPUT3D, MASK3D, BBOX3D, etc.

--- a/kornia/constants.py
+++ b/kornia/constants.py
@@ -90,7 +90,7 @@ class DType(ConstantBase, Enum, metaclass=EnumMetaFlags):
         raise ValueError()
 
 
-# TODO: (low-priority) add INPUT3D, MASK3D, BBOX3D, etc.
+# TODO: (low-priority) add INPUT3D, MASK3D, BBOX3D, LAFs etc.
 class DataKey(ConstantBase, Enum, metaclass=EnumMetaFlags):
     INPUT = 0
     MASK = 1

